### PR TITLE
fix: error msg when query vector dim is wrong

### PR DESCRIPTION
**Pull request by QianZhu**
- changed the error msg for table.search with wrong query vector dim 
- added missing fields for listIndices and indexStats to be consistent with Python API - will make changes in node integ test

### CODE REVIEW

File: rust/lancedb/src/utils.rs [hunk 1]
```diff
@@ -118,14 +118,14 @@ pub(crate) fn default_vector_column(schema: &Schema, dim: Option<i32>) -> Result
         })
         .collect::<Vec<_>>();
     if candidates.is_empty() {
-        Err(Error::Schema {
-            message: "No vector column found to create index".to_string(),
+        Err(Error::InvalidInput {
+            message: "No vector column found to match with the query vector dimension".to_string(),
```
Comment (eddyxu) [inline rust/lancedb/src/utils.rs:122]: can we print the explicit dimension as well?

`format!("No vector column found to match with the query vector dimension: {}", dim.unwrap_or_default())`

File: node/src/index.ts [hunk 2]
```diff
@@ -704,6 +704,9 @@ export interface VectorIndex {
 export interface IndexStats {
   numIndexedRows: number | null
   numUnindexedRows: number | null
+  index_type: string | null
```
Comment (eddyxu) [inline node/src/index.ts:707]: Why are these needed in this PR?

Comment (QianZhu) [reply node/src/index.ts:707]: not related to the vec dim. But just noticed that we did not make changes with node sdk - so wanted to add them

**Review by eddyxu — APPROVED** [review]

File: node/src/remote/index.ts [hunk 3]
```diff
@@ -520,7 +521,10 @@ export class RemoteTable<T = number[]> implements Table<T> {
     const body = await results.body()
     return {
       numIndexedRows: body?.num_indexed_rows,
-      numUnindexedRows: body?.num_unindexed_rows
+      numUnindexedRows: body?.num_unindexed_rows,
+      index_type: body?.index_type,
```
Comment (rpgreen) [inline node/src/remote/index.ts:525]: why are we using a mix of camel case and snake case here?
